src/index.ts

```diff
@@ -501,7 +501,12 @@ class DefaultVisualizer {
         if (getStructures) {
             copy.structures = [] as Structure[];
             for (let i = 0; i < this._dataset.structures.length; i++) {
-                copy.structures.push(this.structure.loadStructure(i, this._dataset.structures[i]));
+                const structure = this.structure.loadStructure(i, this._dataset.structures[i]);
+                // this code path should only be reached with the default loader, that is
+                // synchronous
```

Comment:
I don't think this is the case, this function is public so anyone could call it with `getStructures=true`, even with completely custom loading callbacks.